lib/hammer_cli_foreman/associating_commands.rb

```diff
@@ -236,8 +236,42 @@ class AddProvisioningTemplateCommand < HammerCLIForeman::AddAssociatedCommand
         associated_resource :provisioning_templates
         desc _("Associate a provisioning template")
 
+        option "--provisioning-template-ids", "PROVISIONING_TEMPLATE_IDS", _("List of provisioning template ids"),
+               format: HammerCLI::Options::Normalizers::List.new
+        option "--provisioning-templates", "PROVISIONING_TEMPLATE_NAMES", _("List of provisioning template names"),
+               format: HammerCLI::Options::Normalizers::List.new
+        option "--provisioning-template-search", "PROVISIONING_TEMPLATE_SEARCH", _("Provisioning template name regex to search, all matching templates will be associate")
+
+        def request_params
+          params = super
+          template_ids = params['operatingsystem']['provisioning_template_ids']
+          if options['option_provisioning_template_ids']
+            template_ids |= options['option_provisioning_template_ids']
+          end
+          if options['option_provisioning_templates']
+            options['option_provisioning_templates']&.each do |provisioning_template_name|
```

Comment:
You are also checking for nil on the previous line, I think it is ok to omit one of the checks.

Reply:
Also, we don't use `&.` since this comes with Ruby 2.3+, but hammer could be used on systems with lower version of Ruby.

Honestly, I'd review the minimal supported version of hammer, what do you think about that, @shiramax?